Support -9e as an argument for xz compression

dtcooper opened the pull request:
Hi there,

This allows for a little more extreme compression with `xz` (which is what I believe Raspberry Pi uses for their distributables).

Thanks!

usimd commented on src/pi-gen-config.ts:
```diff
@@ -92,16 +92,17 @@ export async function validateConfig(config: PiGenConfig): Promise<void> {
-  if (!/^[0-9]$/.test(config.compressionLevel)) {
-    throw new Error('compression-level must be between 0 and 9')
+  if (
+    !/^[0-9]$/.test(config.compressionLevel) &&
```
Shouldn't these two conditions be joined using `or`? 
```suggestion
    !/^[0-9]$/.test(config.compressionLevel) ||
```

dtcooper replied:
Pretty sure this is correct

Throw an error when

* (`compressionLevel` does NOT match `^[0-9]$`) and (`deployCompression` is `"xz"` and `compressionLevel` is not `"9e"`)

dtcooper replied:
Woops. I think you're right. Let me take a second look.

usimd reviewed (commented):
Thanks for your addition! Good idea to allow very aggressive compression, didn't even know of this setting 👍

Would you mind adding a test case to also hit the coverage?

dtcooper commented:
> Thanks for your addition! Good idea to allow very aggressive compression, didn't even know of this setting 👍
> 
> Would you mind adding a test case to also hit the coverage?

I'm by no means a typescript expert, but I will try!

dtcooper commented:
Was having trouble writing a test...

dtcooper commented:
Okay! My trouble with the tests were one of the current tests wasn't working properly. It _always_ threw an error because `stageList` wasn't properly converted to a list. I've fixed that test, tested for specific error cases, and added one for the new feature. Thanks!

usimd commented:
Awesome, thanks so much for your effort!

I'll have a closer look tomorrow but I assume this should be fine and we can then merge quickly.

dtcooper commented:
> Awesome, thanks so much for your effort!
> 
> I'll have a closer look tomorrow but I assume this should be fine and we can then merge quickly.

Great! Thanks!

usimd reviewed (approved):
Thanks again for your contribution!
